packages/core/src/local-admin-api.ts

```diff
@@ -32,7 +38,36 @@ export class LocalAdminApi implements AdminApi {
 
   async startIndexingModelData(modelData: Array<ModelData>): Promise<void> {
     await this.indexApi.indexModels(modelData)
-    await this.syncApi.startModelSync(modelData.map((idx) => idx.streamID.toString()))
+
+    if (this.syncApi.enabled) {
+      const models = await Promise.all(modelData.map(({ streamID }) => this.loadStream(streamID)))
```

Comment:
gah I hate that we need to do this circular dependency.  Really I think the way this should work is that when you try to start indexing a set of models, the `core` code should load those models and pass in the necessary information to the indexing code so that the indexing code has everything it needs and doesn't need to load streams.  The reason you can't do that here is that LocalAdminApi and RemoteAdminApi are constrained to having the same interface.  I really think we need to break that requirement and let the `core` code have different APIs than the http client.

CC @PaulLeCam since this is relevant to the refactoring work you are working on.

Reply:
Yeah I agree that having the models injected directly would be better, eventually once we have a proper ComposeDB node we shouldn't need these APIs abstractions/duplication anyway so I'm not too concerned about it.